flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js

```diff
@@ -1262,7 +1265,7 @@ var A320_Neo_UpperECAM;
                         isActive: () => {
                             // Rough approximation until hydraulic system fully implemented
                             // Needs the last 2 conditions because PTU_ON is (incorrectly) permanently set to true during first engine start
-                            return (this.getCachedSimVar("L:XMLVAR_PTU_ON", "Bool") == 1) && (SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent") < 1 || SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent") < 1);
+                            return this.hydPtuMemoVisible.read();
```

Comment:
Would it be possible to just replace the condition, because I don't see the need and understand the reason behind the usage of confirmation and memory nodes. But please do explain your thoughts on this so I may be able to understand you doing so.

`this.getCachedSimVar("L:A32NX_HYD_PTU_ACTIVE_L2R", "bool") || this.getCachedSimVar("L:A32NX_HYD_PTU_ACTIVE_R2L", "bool")`

Reply:
I think the argument in `NXLogic_ConfirmNode` is the time in seconds, not milliseconds, right? I was running into a problem where the HYD PTU message would flicker on and off because the PTU pressurizes in pulses, essentially it pressurizes the system to 3000 psi, then lets off for about a second after which it pressurizes again. To prevent this, I wanted to have the HYD PTU message show immediately when PTU flow was detected but only disappear if it hadn't done anything in 2s. 

Let me know if this can be done maybe with only one ConfirmNode or maybe with none at all?

Reply:
You are totally right, it's in seconds.. my bad.

Reply:
> I think the argument in `NXLogic_ConfirmNode` is the time in seconds, not milliseconds, right? I was running into a problem where the HYD PTU message would flicker on and off because the PTU pressurizes in pulses, essentially it pressurizes the system to 3000 psi, then lets off for about a second after which it pressurizes again. To prevent this, I wanted to have the HYD PTU message show immediately when PTU flow was detected but only disappear if it hadn't done anything in 2s.
> 
> Let me know if this can be done maybe with only one ConfirmNode or maybe with none at all?

Yeah the PTU simbology stays on in the lower ecam HYD page but it pressurizes in pulses like you said, maybe you could use the condition for the lower ecam simbology and copy it?

Reply:
I did actually try and do that, but still ran into the flickering issue. I don't know why this doesn't happen on the lower ECAM to be honest.

Reply:
There is a side effect on this fix: During the second engine start, the PTU self-tests itself (the famous barking sound in the cabin) during this PTU activation the HYD PTU ecam memo should be inhibited but it briefly apears as the system build pressure and deactivates it, it's worth noting that in the sim that PTU is not actually doing it's self test, but activating as soon as you set the second engine master to on since it detecs that the engine is on but there is no hyd pressure on that side, might be a deeper hyd system issue not yet fesible to implement a fix, but if possible, inhibit the HYD PTU message until the second engine finishes starting

Reply:
I think there is no actual "PTU self test" per say as far as I understand. When starting the second engine, it's just supposed to activate PTU valve, and computer just waits for pressure to rise to 3000 in less than X seconds to validate PTU test (as the engine driven pump is not supposed to provide enough flow on the very begining of engine spool up). I don't know if there is a specific condition to inhibit the message on upper ecam though.

Reply:
I think their point is simply that, while it is correct for the PTU to come on at this stage, the HYD PTU memo should not appear on the ECAM. I am not sure how I should inhibit this. Checking if both engines are available is not correct since the PTU should come on in the event of an engine failure.

Reply:
Just checked the logic and there's actually no inhibit condition. If edp pressure is low on one side, and system pressure is >1450 on both sides it should show basically

Reply:
> Just checked the logic and there's actually no inhibit condition. If edp pressure is low on one side, and system pressure is >1450 on both sides it should show basically

Like I explained, the PTU self-test itself on second engine start, but it does not show the HYD PTU memo during the test or the engines start as a whole

https://www.youtube.com/watch?v=L94f0ad9x3Q

Reply:
you could use the FWC flight phase to determine whether to actually show the message or not.